apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/BranchController.java

```diff
@@ -0,0 +1,139 @@
+package com.ctrip.framework.apollo.openapi.v1.controller;
+
+/**
+ * Created by qianjie on 8/10/17.
+ */
+
+import com.ctrip.framework.apollo.common.dto.GrayReleaseRuleDTO;
+import com.ctrip.framework.apollo.common.dto.NamespaceDTO;
+import com.ctrip.framework.apollo.common.dto.ReleaseDTO;
+import com.ctrip.framework.apollo.common.exception.BadRequestException;
+import com.ctrip.framework.apollo.core.enums.Env;
+import com.ctrip.framework.apollo.portal.component.PermissionValidator;
+import com.ctrip.framework.apollo.portal.component.config.PortalConfig;
+import com.ctrip.framework.apollo.portal.entity.bo.NamespaceBO;
+import com.ctrip.framework.apollo.portal.entity.model.NamespaceReleaseModel;
+import com.ctrip.framework.apollo.portal.listener.ConfigPublishEvent;
+import com.ctrip.framework.apollo.portal.service.NamespaceBranchService;
+import com.ctrip.framework.apollo.portal.service.ReleaseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestController
+@RequestMapping("/openapi/v1/envs/{env}")
+public class BranchController {
+
+    @Autowired
+    private PermissionValidator permissionValidator;
+    @Autowired
+    private ReleaseService releaseService;
+    @Autowired
+    private NamespaceBranchService namespaceBranchService;
+    @Autowired
+    private ApplicationEventPublisher publisher;
+    @Autowired
+    private PortalConfig portalConfig;
+
+    @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches", method = RequestMethod.GET)
+    public NamespaceBO findBranch(@PathVariable String appId,
```

Comment:
Normally we would use another set of dto for open api, so that any internal change won't affect outside users. So please use open api dto, just as [Open API ReleaseController](https://github.com/ctripcorp/apollo/blob/master/apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/ReleaseController.java).

BTW, in the current master branch, we have extracted open api dto to another module [apollo-openapi](https://github.com/ctripcorp/apollo/tree/master/apollo-openapi/src/main/java/com/ctrip/framework/apollo/openapi/dto), you may first update your local code base first.